bootloader/src/pyi_launch.c

```diff
@@ -417,14 +422,51 @@ pyi_launch_run_scripts(ARCHIVE_STATUS *status)
             /* Run it */
             retval = PI_PyEval_EvalCode(code, main_dict, main_dict);
 
-            /* If retval is NULL, an error occured. Otherwise, it is a Python object.
+            /* If retval is NULL, an error occurred. Otherwise, it is a Python object.
              * (Since we evaluate module-level code, which is not allowed to return an
              * object, the Python object returned is always None.) */
             if (!retval) {
-                PI_PyErr_Print();
-                /* If the error was SystemExit, PyErr_Print calls exit() without
-                 * returning. So don't print "Failed to execute" on SystemExit. */
                 FATALERROR("Failed to execute script %s\n", ptoc->name);
+
+                #if defined(_WIN32) && defined(WINDOWED) && defined(LAUNCH_DEBUG)
+                    /* If running in Windows in windowed mode with debug,
+                     * print the traceback to a message box */
+
+                    /* First get the value of the error */
+                    PI_PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+                    pvalue_str = PI_PyObject_Str(pvalue);
+                    pvalue_cchar = PI_PyUnicode_AsUTF8(pvalue_str);
+                    FATALERROR("Error: %s\n", pvalue_cchar);
+                    Py_DECREF(pvalue_str);
+
+                    /* Next see if we can get a full traceback, only works
+                     * with --noarchive */
```

Comment:
I assume this should read: "Do show the source line only with --noarchive", correct?

Reply:
I was trying to say that this traceback will only be able to be collected if running in PyInstaller with the --noarchive option.

Reply:
So this does not display any traceback at all? Not even one like thus (not showing the source):
```
[hartmut@thisbe x]$ ./x 
Traceback (most recent call last):
  File "x.py", line 12, in <module>
  File "x.py", line 8, in f3
  File "x.py", line 6, in f2
  File "x.py", line 4, in f1
ZeroDivisionError: division by zero
```